lib/graphql/metrics/instrumentation.rb

```diff
@@ -2,68 +2,83 @@
+    module Instrumentation
+      def initialize(processor_class:, **options)
+        query_or_multiplex = options[:query] || options[:multiplex]
+        @processor = processor_class.new(query_or_multiplex)
+        super
+      end
+
+      def execute_multiplex(multiplex:)
+        return super if multiplex.context[GraphQL::Metrics::SKIP_GRAPHQL_METRICS_ANALYSIS]
+
+        result = nil
+
+        multiplex.queries.each do |query|
+          ns = query.context.namespace(CONTEXT_NAMESPACE)
+          ns[GraphQL::Metrics::INLINE_FIELD_TIMINGS] = Hash.new { |h, k| h[k] = [] }
+          ns[GraphQL::Metrics::LAZY_FIELD_TIMINGS] = Hash.new { |h, k| h[k] = [] }
         end
 
-        # Even if queries are present and valid, applications may set this context value in order to opt out of
-        # having Analyzer and Tracer gather runtime metrics.
-        # If we're skipping runtime metrics, then both Instrumentation before_ and after_query can and should be
-        # short-circuited as well.
-        return if query.context[GraphQL::Metrics::SKIP_GRAPHQL_METRICS_ANALYSIS]
+        begin
+          result = super
+        ensure
+          multiplex.queries.each do |query|
+            handle_query(query)
+          end
+        end
 
-        ns = query.context.namespace(CONTEXT_NAMESPACE)
-        ns[GraphQL::Metrics::TIMINGS_CAPTURE_ENABLED] = true
-        ns[GraphQL::Metrics::INLINE_FIELD_TIMINGS] = {}
-        ns[GraphQL::Metrics::LAZY_FIELD_TIMINGS] = {}
+        result
       end
 
-      def after_query(query)
-        return if query.context[GraphQL::Metrics::SKIP_GRAPHQL_METRICS_ANALYSIS]
+      private
 
+      def handle_query(query)
         ns = query.context.namespace(CONTEXT_NAMESPACE)
-        analyzer = ns[GraphQL::Metrics::ANALYZER_INSTANCE_KEY]
-
-        if runtime_metrics_interrupted?(ns)
-          # If runtime metrics were interrupted, then it's most likely that the application raised an exception and that
-          # query parsing (which is instrumented by GraphQL::Metrics::Tracer) was abruptly stopped.
-          #
-          # In this scenario, we still attempt to log whatever static query metrics we've collected, with runtime
-          # metrics (like query, field resolver timings) excluded.
-          analyzer.extract_fields(with_runtime_metrics: false)
-          analyzer.extract_query
-        else
+
+        query_metrics = ns[:query_metrics].to_h
+
+        unless runtime_metrics_interrupted?(ns)
           query_duration = GraphQL::Metrics.current_time_monotonic - ns[GraphQL::Metrics::QUERY_START_TIME_MONOTONIC]
 
           runtime_query_metrics = {
             query_start_time: ns[GraphQL::Metrics::QUERY_START_TIME],
             query_duration: query_duration,
             parsing_duration: ns[GraphQL::Metrics::PARSING_DURATION],
             validation_duration: ns[GraphQL::Metrics::VALIDATION_DURATION],
-            lexing_duration: ns[GraphQL::Metrics::LEXING_DURATION],
```

Comment:
Should we keep this? It is still possible to get a lex duration depending on the parser used.

https://github.com/rmosolgo/graphql-ruby/blob/88b4c0ac1d7af7ad7cfc262d2f8fb5face3555fd/lib/graphql/tracing/trace.rb#L18-L21

Reply:
Ok. For our internal stuff, I kept tests for both cases, but I do like the simplification as this is the default now.